### DISCUSSION TITLE
Failed to Build BaseTools/Edk2ToolsBuild.py

### DISCUSSION THREAD

**Discussion by mmfahrrestatp** [tianocore/edk2#4611] [Q&A]
When I run this script:
`python BaseTools/Edk2ToolsBuild.py -t VS2019`

I got this log and error message
```
$ python BaseTools/Edk2ToolsBuild.py -t VS2019
SECTION - Init Self Describing Environment
SECTION - Loading Plugins
SECTION - Start Invocable Tool
Traceback (most recent call last):
  File "C:\Users\Aaa\y\apps\src\edk2\BaseTools\Edk2ToolsBuild.py", line 177, in <module>
    main()
  File "C:\Users\Aaa\y\apps\src\edk2\BaseTools\Edk2ToolsBuild.py", line 173, in main
    Edk2ToolsBuild().Invoke()
  File "C:\Users\Aaa\AppData\Local\Programs\Python\Python311\Lib\site-packages\edk2toolext\base_abst
ract_invocable.py", line 236, in Invoke
    retcode = self.Go()
              ^^^^^^^^^
  File "C:\Users\Aaa\y\apps\src\edk2\BaseTools\Edk2ToolsBuild.py", line 139, in Go
    raise Exception("Failed to build.")
Exception: Failed to build.
```
I'm following from this step
https://github.com/tianocore/edk2/blob/master/OvmfPkg/PlatformCI/ReadMe.md
and stuck at building BaseTools

Run this script from Windows 10 with Git Bash Terminal
Python version python 3.11.3
Tool chain tag = VS2019

Had checked this page
https://github.com/tianocore/edk2-pytool-extensions/issues/207
https://github.com/tianocore/edk2-pytool-extensions/issues/180

But this command
`git submodule update --init --recursive`
still didn't solve this error.

I'm new about this, thanks for anyone support

**Comment by Flickdm** [comment]
That particular error indicates that there was a make build error.

You can review the build logs at:

```
$(workspace)/BaseTools/BaseToolsBuild/BASETOOLS_BUILD.txt
```
to review the make error that is being bubbled up.

This should help you figure out the exact reason for the error
